control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go

```diff
@@ -306,15 +306,36 @@ func (r *HostedControlPlaneReconciler) Reconcile(ctx context.Context, req ctrl.R
 
 	// Return early if deleted
 	if !hostedControlPlane.DeletionTimestamp.IsZero() {
-
+		condition := &metav1.Condition{
+			Type: string(hyperv1.AWSDefaultSecurityGroupDeletion),
+		}
 		if shouldCleanupCloudResources(r.Log, hostedControlPlane) {
 			if err := r.destroyAWSDefaultSecurityGroup(ctx, hostedControlPlane); err != nil {
+				condition.Message = err.Error()
```

Comment:
If we are going to include the error message here, we need to ensure that the `destroyAWSDefaultSecurityGroup` function does not include a direct AWS error in any of the errors it returns. Otherwise, we will potentially get a different error message with every reconcile and enter a hot reconcile loop

Reply:
yep, the function `destroyAWSDefaultSecurityGroup` is not returning any direct error from AWS, all the messages are handcrafted but variables inside of the error itself.

Reply:
What about something like this? https://github.com/jparrill/hypershift/blob/41e3cf964e6921f93f2009f04c922078b94fc3aa/control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go#L4176